packages/compiler-sfc/src/templateTransformAssetUrl.ts

```diff
@@ -1,5 +1,46 @@
-import { NodeTransform } from '@vue/compiler-core'
+import {
+  AttributeNode,
+  createSimpleExpression,
+  NodeTransform,
+  NodeTypes
+} from '@vue/compiler-core'
+import { parseUrl } from './templateUtils'
 
-export const transformAssetUrl: NodeTransform = () => {
-  // TODO
+export interface AssetURLOptions {
+  [name: string]: string[]
+}
+
+const assetURLOptions: AssetURLOptions = {
+  video: ['src', 'poster'],
+  source: ['src'],
+  img: ['src'],
+  image: ['xlink:href', 'href'],
+  use: ['xlink:href', 'href']
+}
+
+export const transformAssetUrl: NodeTransform = (node, context) => {
+  if (node.type === NodeTypes.ELEMENT) {
+    for (const tag in assetURLOptions) {
+      if ((tag === '*' || node.tag === tag) && node.props.length) {
+        const attributes = assetURLOptions[tag]
+        attributes.forEach(item => {
+          node.props.forEach((attr: AttributeNode, index) => {
+            if (attr.type !== NodeTypes.ATTRIBUTE) return
+            if (attr.name !== item) return
+            if (!attr.value) return
+            const url = parseUrl(attr.value.content)
+            const exp = context.file(url.path, url.hash, attr.loc)
```

Comment:
I think `context.file`'s implementation is too coupled to this specific use case. In other words, current logic in `context.file` (which is in `compiler-core`) should be here (in `compiler-sfc`) instead.

Instead of `context.file`, it could be something more generic like `context.imports`:

``` ts
interface TransformContext {
  // ...
  imports: Set<{ var: string | ExpressionNode, path: string }>
}
```

Then in here:

``` js
const fileIndex = context.imports.size
const exp = /* ... previous context.file logic */
context.imports.add({ var: exp, path: url.path })
```